src/algokit/cli/tasks/mint.py

```diff
@@ -0,0 +1,163 @@
+import logging
+import math
+from pathlib import Path
+
+import click
+
+from algokit.cli.tasks.utils import (
+    get_account_with_private_key,
+    get_asset_explorer_url,
+    get_transaction_explorer_url,
+    load_algod_client,
+)
+from algokit.core.tasks.ipfs import get_web3_storage_api_key
+from algokit.core.tasks.mint.mint import MetadataStandard, mint_token
+from algokit.core.tasks.mint.models import TokenMetadata
+
+logger = logging.getLogger(__name__)
+
+
+def _validate_inputs(total: int, decimals: int) -> None:
+    # Validate total and decimals
+    if not (total == 1 or (total % 10 == 0 and total != 0)):
+        raise click.ClickException("Total must be 1 or a power of 10 larger than 1 (10, 100, 1000, ...).")
+    if not ((total == 1 and decimals == 0) or (total != 1 and decimals == int(math.log10(total)))):
```

Comment:
This is true for a fractional NFT, but for a fungible token total and decimals can be whatever right?

Reply:
![CleanShot 2023-10-16 at 19 12 41](https://github.com/algorandfoundation/algokit-cli/assets/7698600/9fd25ff3-ad5e-457f-bf20-a351b10d1b5a)
I primarily based this of this chunk in https://arc.algorand.foundation/ARCs/arc-0003, assumed this implies that ARC3 spec is only for NFTs (fractional or not)

Reply:
@robdmoore added extra flag --nft/--ft for checking when above validators need to be checked, if --ft is set then anything in total/decimals is accepted